controllers/feeds/getFeedChat.js

```diff
@@ -205,6 +208,17 @@ const getFeedChatService = async (req, res) => {
       newFeed.push(...mapping);
     }
 
+    let actors = [];
+    for (const feed of newFeed) {
+      if (!actors.includes(feed.actor.id)) {
+        actors.push(feed.actor.id);
+      }
+      if (feed.reaction && !actors.includes(feed.reaction.user_id)) {
+        actors.push(feed.reaction.user_id);
+      }
+    }
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code collects all unique actor IDs from the `newFeed` array. It then uses these IDs to fetch the corresponding karma scores. This is an efficient way to ensure that only necessary data is fetched from the database.
```diff
+ let actors = [];
+ for (const feed of newFeed) {
+   if (!actors.includes(feed.actor.id)) {
+     actors.push(feed.actor.id);
+   }
+   if (feed.reaction && !actors.includes(feed.reaction.user_id)) {
+     actors.push(feed.reaction.user_id);
+   }
+ }
+ const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->